weave-js/src/util/date.ts

```
@@ -0,0 +1,387 @@
+  }
+
+  // Handle shorthand relative dates (e.g., "1d", "2w", "3m", "1y")
+  const shorthandPattern = /^(\d+)([dwmy])$/i;
+  const shorthandMatch = trimmedStr.match(shorthandPattern);
+  if (shorthandMatch) {
+    const amount = parseInt(shorthandMatch[1], 10);
+    const unit = shorthandMatch[2].toLowerCase();
+
+    switch (unit) {
+      case 'd':
+        const daysAgo = new Date(now);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)